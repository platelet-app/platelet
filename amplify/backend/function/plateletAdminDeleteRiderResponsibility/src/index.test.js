const { handler } = require("./index");

jest.mock(
    "/opt/graphql/queries",
    () => {
        return {
            listPossibleRiderResponsibilities:
                "listPossibleRiderResponsibilities",
        };
    },
    { virtual: true }
);
jest.mock(
    "/opt/graphql/mutations",
    () => {
        return {
            deleteRiderResponsibility: "deleteRiderResponsibility",
            deletePossibleRiderResponsibility:
                "deletePossibleRiderResponsibility",
        };
    },
    { virtual: true }
);

const errorCheck = (body) => {
    if (body?.errors) {
        console.error(body?.errors);
        throw new Error(body?.errors[0].message);
    }
};
jest.mock(
    "/opt/appSyncRequest",
    () => {
        return {
            request: jest.fn(),
            errorCheck,
        };
    },
    { virtual: true }
);

const appsyncModule = require("/opt/appSyncRequest");

const mockRiderResponsibility = {
    _version: 1,
    _deleted: null,
    _lastChangedAt: 1620000000000,
};

const possibleRiderResponsibilitiesFakeData = {
    listPossibleRiderResponsibilities: {
        items: [
            { id: "someId1", _version: 1, _deleted: false },
            { id: "someId2", _version: 10, _deleted: false },
        ],
        nextToken: null,
    },
};

const fakeRiderResponsibility = {
    getRiderResponsibility: {
        id: "someRiderRespId",
        _version: 5,
        _deleted: false,
    },
};

const possibleRiderResponsibilityDeleted = {
    deleteRiderResponsibility: {
        id: "someId1",
        _version: 2,
        _deleted: true,
    },
};
const possibleRiderResponsibilityDeleted2 = {
    deleteRiderResponsibility: {
        id: "someId2",
        _version: 2,
        _deleted: true,
    },
};

function setupFetchStub(data) {
    return function fetchStub(_url) {
        return new Promise((resolve) => {
            resolve({
                json: () =>
                    Promise.resolve({
                        data,
                    }),
            });
        });
    };
}

describe("index", () => {
    test("delete a rider responsibility and associated possible rider responsibilities", () => {
        appsyncModule.request
            .mockImplementationOnce(
                setupFetchStub(possibleRiderResponsibilitiesFakeData)
            )
            .mockImplementationOnce(
                setupFetchStub(possibleRiderResponsibilityDeleted)
            )
            .mockImplementationOnce(
                setupFetchStub(possibleRiderResponsibilityDeleted2)
            )
            .mockImplementationOnce(setupFetchStub(fakeRiderResponsibility))
            .mockImplementationOnce(
                setupFetchStub({ ...fakeRiderResponsibility, _deleted: true })
            );
        const event = {
            arguments: { riderResponsibilityId: "someRiderRespId" },
        };
        handler(event);
    });
});
