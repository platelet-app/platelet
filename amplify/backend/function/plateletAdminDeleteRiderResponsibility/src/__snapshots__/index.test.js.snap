// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`index delete a rider responsibility and associated possible rider responsibilities 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "query": "listPossibleRiderResponsibilities",
        "variables": {
          "filter": {
            "_deleted": {
              "ne": true,
            },
            "riderResponsibilityPossibleUsersId": {
              "eq": "someRiderRespId",
            },
          },
          "nextToken": null,
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 1,
            "id": "someId1",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 10,
            "id": "someId2",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "
    query GetRiderResponsibility($id: ID!) {
        getRiderResponsibility(id: $id) {
            id
            _version
            _deleted
        }
    }
",
        "variables": {
          "id": "someRiderRespId",
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deleteRiderResponsibility",
        "variables": {
          "input": {
            "_version": 5,
            "id": "someRiderRespId",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`index paginated rider responsibilities 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "query": "listPossibleRiderResponsibilities",
        "variables": {
          "filter": {
            "_deleted": {
              "ne": true,
            },
            "riderResponsibilityPossibleUsersId": {
              "eq": "someRiderRespId",
            },
          },
          "nextToken": null,
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "listPossibleRiderResponsibilities",
        "variables": {
          "filter": {
            "_deleted": {
              "ne": true,
            },
            "riderResponsibilityPossibleUsersId": {
              "eq": "someRiderRespId",
            },
          },
          "nextToken": "someToken",
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 3,
            "id": "someId3",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 4,
            "id": "someId4",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 1,
            "id": "someId1",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deletePossibleRiderResponsibilities",
        "variables": {
          "input": {
            "_version": 10,
            "id": "someId2",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "
    query GetRiderResponsibility($id: ID!) {
        getRiderResponsibility(id: $id) {
            id
            _version
            _deleted
        }
    }
",
        "variables": {
          "id": "someRiderRespId",
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "deleteRiderResponsibility",
        "variables": {
          "input": {
            "_version": 5,
            "id": "someRiderRespId",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
